Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/Auth.hs

```diff
@@ -79,40 +82,45 @@ verifyAdmin user = do
     throwError AccessDenied
   return user
 
-verifyToken :: (HasEsqEnv m r, MonadFlow m) => RegToken -> m SR.RegistrationToken
+verifyToken :: (HasEsqEnv m r, MonadFlow m, Utils.EsqDBFlow m r, Utils.CacheFlow m r) => RegToken -> m SR.RegistrationToken
 verifyToken regToken = do
   QR.findByToken regToken
     >>= Utils.fromMaybeM (InvalidToken regToken)
     >>= validateToken
 
-validateAdmin :: (HasEsqEnv m r, EncFlow m r) => RegToken -> m Person.Person
+validateAdmin :: (HasEsqEnv m r, EncFlow m r, Utils.EsqDBFlow m r, Utils.CacheFlow m r) => RegToken -> m Person.Person
 validateAdmin regToken = do
   SR.RegistrationToken {..} <- verifyToken regToken
   user <-
     QP.findById (Id entityId)
       >>= fromMaybeM (PersonNotFound entityId)
   verifyAdmin user
 
-verifyPerson :: (HasEsqEnv m r, Redis.HedisFlow m r, HasField "authTokenCacheExpiry" r Seconds, MonadFlow m) => RegToken -> m (Id Person.Person, Id Merchant.Merchant)
+verifyPerson :: (HasEsqEnv m r, Redis.HedisFlow m r, HasField "authTokenCacheExpiry" r Seconds, MonadFlow m, Utils.EsqDBFlow m r, Utils.CacheFlow m r) => RegToken -> m (Id Person.Person, Id Merchant.Merchant, Id DMOC.MerchantOperatingCity)
 verifyPerson token = do
   let key = authTokenCacheKey token
   authTokenCacheExpiry <- getSeconds <$> asks (.authTokenCacheExpiry)
   result <- Redis.safeGet key
   case result of
-    Just (personId, merchantId) -> return (personId, merchantId)
+    Just (personId, merchantId, Nothing) -> do
```

Comment:
Nope this will not work. It should be exactly (personId, merchantId)

Reply:
Changed